src/prefect/tasks/sftp/sftp.py

```diff
@@ -0,0 +1,280 @@
+        - port_number (int): the port number to connect to the server.
+        - remote_path (str): the remote sftp file path
+        - local_path (str): the local file path to download file to
+        - **kwargs (dict, optional): additional keyword arguments to pass to the Task constructor.
+
+    Raises:
+        - ValueError: if a required parameter is not supplied.
+        - ClientError: if exception occurs when connecting/downloading from the server.
+    """
+
+    def __init__(
+        self,
+        host: str = None,
+        username: str = None,
+        password: str = None,
+        port_number: int = None,
+        remote_path: str = None,
+        local_path: str = None,
+        **kwargs,
+    ):
+        self.host = host
+        self.username = username
+        self.password = password
+        self.port_number = port_number
+        self.remote_path = remote_path
+        self.local_path = local_path
+        super().__init__(**kwargs)
+
+    def _create_connection(self) -> None:
+        """
+        Initialise the connection with the SFTP server
+        :return: None
+        """
+        transport = Transport(sock=(self.host, self.port_number))
+        transport.connect(username=self.username, password=self.password)
+        self._connection = SFTPClient.from_transport(transport)
+        self.logger.info(f"connected to {self.host}, {self.port_number}")
+
+    def file_exists(self, remote_path: str) -> bool:
+        """
+        Checks if file exists in remote path or not.
+
+        Args:
+            - remote_path (str): checks if the specified remote
+            file path exists on the sftp server
+
+        """
+        res: bool = True
+        try:
+            self.logger.info("remote path : ", remote_path)
+            self._connection.stat(remote_path)
+            res = True
+        except IOError as e:
+            if e.errno == errno.ENOENT:
+                res = False
+            raise
+        finally:
+            return res
+
+    def download(self, remote_path: str, local_path: str, retry: int = 5) -> bool:
+        """
+        Downloads file/files from the specified remote_path to a local_path
+
+        Args:
+            - remote_path (str): the remote sftp file path
+            - local_path (str): the local file path to download file to
+            - retry (int): number of retries for download process
+
+        """
+        # first check if local path exists or not
+        local_dir = "/".join(local_path.split("/")[:-1]) + "/"
+        if not os.path.isdir(local_dir):
+            os.mkdir(local_dir)
+
+        # check remote_path
+        if self.file_exists(remote_path) or retry == 0:
+            self._connection.get(remote_path, local_path, callback=None)
+            return True
+        elif retry > 0:
+            time.sleep(5)
+            retry = retry - 1
+            self.download(remote_path, local_path, retry=retry)
+
+        return False
```

Comment:
The way this function is written, it will always return false. I'd suggest raising an exception if the download fails after the configured number of retries.